packages/solana-wallet-injection/src/wallet.ts

```diff
@@ -0,0 +1,184 @@
+    const { accepted } = await this.enable();
+
+    if (!accepted) {
+      throw new Error('User rejected');
+    }
+
+    const addresses = await this.callMethod(COMMON_REQUEST_MAP.wallet_getAddresses);
+    const [{ publicKey }] = addresses;
+    this.publicKey = new PublicKey(publicKey);
+    this.isConnected = true;
+
+    console.debug(`Public Key`, this.publicKey);
+    console.debug(`Public Key Decoded`, this.publicKey.toBase58());
+    console.log('Is Connected: ', this.isConnected);
+    return { publicKey };
+  }
+
+  public async disconnect(): Promise<void> {
+    this.publicKey = null;
+    this.isConnected = false;
+  }
+
+  public async signAndSendTransaction<T extends Transaction | VersionedTransaction>(
+    transaction: T,
+    options?: SendOptions | undefined
+  ): Promise<{ signature: string }> {
+    console.debug('signAndSendTransaction');
+    const signedRawTx = (await this.signTransaction(transaction)).serialize();
+    console.debug('signAndSendTransaction: signedRawTx', signedRawTx);
+    const signature = await this.connection.sendRawTransaction(signedRawTx, options);
+    console.debug('signAndSendTransaction: signature', signature);
+    return { signature };
+  }
+
+  public async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<any> {
+    console.debug('signTransaction');
+    const instructions = this.getTransactionDetails(transaction);
+    const secretKeyBase58 = await this.callMethod(COMMON_REQUEST_MAP.wallet_getSigner, {
+      transaction,
+      instructions,
+      fee: DEFAULT_FEE,
+    });
+    return this._signTransaction(transaction, secretKeyBase58);
+  }
+
+  public async signAllTransactions<T extends Transaction | VersionedTransaction>(transactions: T[]): Promise<T[]> {
+    console.debug('signAllTransactions');
+    const instructions: ParsedInstruction<Idl, string>[] = [];
+    const signedTransactions: T[] = [];
+
+    for (const t of transactions) {
+      const _instructions = this.getTransactionDetails(t);
+      instructions.push(..._instructions);
+    }
+
+    const secretKeyBase58 = await this.callMethod(COMMON_REQUEST_MAP.wallet_getSigner, {
+      transactions,
+      instructions,
+      fee: new BigNumber(transactions.length).multipliedBy(DEFAULT_FEE).toString(),
+    });
+
+    for (const t of transactions) {
+      const signedTx = this._signTransaction(t, secretKeyBase58);
+      signedTransactions.push(signedTx);
+    }
+    return signedTransactions;
+  }
+
+  // TODO: to be tested
+  public async signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }> {
+    console.debug('signMessage', message);
+    console.debug('signMessage: string', message.toString());
+    return this.callMethod(COMMON_REQUEST_MAP.wallet_signMessage, message.toString());
+  }
+
+  public on<E extends keyof LiqualityEvent>(_event: E, _listener: LiqualityEvent[E], _context?: any): void {
+    console.debug('on', _event);
```

Comment:
So there are event listeners part of the spec. Are we implementing it? If not, what is going to be incompatible?

Reply:
I don't know. 
I'm testing here as stated in their official guide:
https://solana-labs.github.io/wallet-adapter/example/

it seems to work

Reply:
Ok, if it's not part of the test harness, then I guess we wait and see if it gets utilised.